src/main/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -81,20 +81,17 @@ class WorldConfiguration {
      * @param compFactory the constructor method for creating the component.
      * @param listenerFactory the constructor method for creating the component listener.
      * @throws [FleksComponentAlreadyAddedException] if the component was already added before.
-     * @throws [FleksComponentListenerAlreadyAddedException] if the listener was already added before.
      * @throws [FleksInjectableTypeHasNoName] if the dependency type has no T::class.simpleName.
      */
-    inline fun <reified T : Any> component(noinline compFactory: () -> T, noinline listenerFactory: (() -> ComponentListener<*>)? = null) {
+    inline fun <reified T : Any> component(noinline compFactory: () -> T, noinline listenerFactory: (() -> ComponentListener<T>)? = null) {
         val compType = T::class.simpleName ?: throw FleksInjectableTypeHasNoName(T::class)
 
         if (compType in componentFactory) {
             throw FleksComponentAlreadyAddedException(compType)
         }
         componentFactory[compType] = compFactory
         if (listenerFactory != null) {
-            if (compType in compListenerFactory) {
-                throw FleksComponentListenerAlreadyAddedException(compType)
-            }
+            // No need to check compType again in compListenerFactory - it is already guarded with check in componentFactory
```

Comment:
While writing/checking the unit test I realized that this check is not needed since the listener type cannot be different to the component type. Otherwise there will be already a compile error. This is nice since the user cannot do anything wrong when registering the component together with the listener object.